File: src/main/java/komu/demodel/domain/TypeType.java
/*
 * Copyright (C) 2010 Juha Komulainen. All rights reserved.
 */
package komu.demodel.domain;

public enum TypeType {
    CONCRETE_CLASS, ABSTRACT_CLASS, INTERFACE;
    
    public boolean isAbstract() {
        return this != CONCRETE_CLASS;
    }
}

File: src/main/scala/komu/demodel/domain/TypeType.scala
/*
 * Copyright (C) 2010 Juha Komulainen. All rights reserved.
 */
package komu.demodel.domain

sealed abstract class TypeType {
  def isAbstract = this != TypeType.CONCRETE_CLASS
}

object TypeType {
  case object CONCRETE_CLASS extends TypeType
  case object ABSTRACT_CLASS extends TypeType
  case object INTERFACE extends TypeType
}
